src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -1837,23 +1864,67 @@ public void setSilentMode(boolean silentMode) {
         this.silentMode = silentMode;
     }
 
+    /**
+     * DataBoundSetter for TopicAssociation.
+     * Used for jelly file.
+     *
+     * @param topicAssociation the TopicAssociation object.
+     */
+    @DataBoundSetter
+    public void setTopicAssociation(final TopicAssociation topicAssociation) {
+        this.topicAssociation = topicAssociation;
+    }
+
+    /**
+     * Enable or disable Topic Association option.
+     *
+     * @param enable true or false.
+     */
+    public void setEnableTopicAssociation(boolean enable) {
+        if (this.topicAssociation == null) {
+            this.topicAssociation = new TopicAssociation();
+        }
+        this.topicAssociation.setEnabled(enable);
+    }
+
     /**
      * Check if topic association is enabled.
+     * Used for jelly file.
+     *
      * @return true if so.
      */
-    public boolean isEnableTopicAssociation() {
-        return enableTopicAssociation;
+    public boolean isTopicAssociation() {
+        return topicAssociation != null && topicAssociation.isEnabled();
     }
 
     /**
-     * Set topic association enabled.
-     * When topic association is on the job can be triggered by project not in its configuration
-     * as long as there are projects of interest in the same topic.
-     * @param enableTopicAssociation true if it should be enabled.
+     * Returns true if a change in state NEW should be ignored otherwise false.
+     * Used for jelly file.
+     *
+     * @return true if it should be ignored otherwise false
      */
-    @DataBoundSetter
-    public void setEnableTopicAssociation(boolean enableTopicAssociation) {
```

Comment:
same here, just mark them as `@Deprecated` if it is not intended to be used any more. `@DataBoundSetter` might be removed, i'm not sure if that would break any JCasC scenarios or not.